pyccel/complexity/memory.py

```diff
@@ -10,228 +10,187 @@
-        f = d['f']
-        m = d['m']
-        # ...
-
-        # ...
-        lt_f = leading_term(f, *args)
-        lt_m = leading_term(m, *args)
-
-        q = lt_f/lt_m
-        # ...
-
-        # ...
-        if verbose:
-            print((" arithmetic cost         ~ " + str(f)))
-            print((" memory cost             ~ " + str(m)))
-            print((" computational intensity ~ " + str(q)))
-        # ...
-
-        return q
+            # ...
+            step = 1
+            if not i.step == None:
+                if isinstance(i.step, Literal):
+                    step = i.step.python_value
+                else:
+                    step = pyccel_to_sympy(i.step, self._symbol_map, self._used_names)
+            # ...
+
+            if not(step == 1):
+                raise NotImplementedError('only step == 1 is treated')
+
+            # TODO uncomment this
+            #      this was commented because we get floor(...)
+            ntimes *= (stop - start) #// step
+
+        return ntimes * READ
+
+    def _cost_Allocate(self, expr, **settings):
+        # TODO
+        return 0
+
+    def _cost_Deallocate(self, expr, **settings):
+        # TODO
+        return 0
+
+    def _cost_NumpyFloor(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyExp(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyLog(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpySqrt(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpySin(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyCos(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyTan(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyArcsin(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyArccos(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyArctan(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyArctan2(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpySinh(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyCosh(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyTanh(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
+
+    def _cost_NumpyArcsinh(self, expr, **settings):
+        return sum(self._cost(i, **settings) for i in expr.args)
```

Comment:
If all these Numpy functions do the same thing, you should be able to simplify it to something like:
```
def _cost_NumpyUfuncBase(self, expr, **settings):
        return sum(self._cost(i, **settings) for i in expr.args
```
(even if 1 or 2 do something different you can just specialise these cases)